### PR TITLE
stats: Skip empty symbolic segments in stat_merger_fuzz_test

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message: Found another corner-case that doesn't make sense, where a diff is registered.
Additional Description:
Risk Level: low
Testing: just the one fuzz test
Docs Changes:
Release Notes:
Platform Specific Features:

Fixes
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=32294

### CODE REVIEW

File: test/common/stats/stat_merger_fuzz_test.cc [hunk 1]
```diff
@@ -54,6 +54,9 @@ void testDynamicEncoding(absl::string_view data, SymbolTable& symbol_table) {
     if (is_symbolic) {
       absl::string_view::size_type pos = segment.find_first_not_of('.');
       segment.remove_prefix((pos == absl::string_view::npos) ? segment.size() : pos);
+      if (segment.empty()) {
```
Comment (asraa) [inline test/common/stats/stat_merger_fuzz_test.cc:57]: could you add a short testcase that triggers this path?

File: test/common/stats/stat_merger_fuzz_test.cc [hunk 2]
```diff
@@ -74,6 +77,15 @@ void testDynamicEncoding(absl::string_view data, SymbolTable& symbol_table) {
   }
   StatName decoded = dynamic_context.makeDynamicStatName(name, dynamic_map);
   std::string decoded_str = symbol_table.toString(decoded);
+  uint32_t max_messages = 10;
```
Comment (asraa) [inline test/common/stats/stat_merger_fuzz_test.cc:80]: is this just for debugging?

**Review by asraa — COMMENTED** [review]
Thanks!

File: test/common/stats/stat_merger_fuzz_test.cc (hunk 1 above)
Comment (jmarantz) [reply test/common/stats/stat_merger_fuzz_test.cc:57]: Actually this is really a question of whether we should include empty segments when encoding.

Coincidentally there's an in-progress fix for this from @jessicayuen : https://github.com/envoyproxy/envoy/compare/main...jessicayuen:az-stat-fix and if that lands then the testcase that was failing starts to pass.

Note that we are have unit-tests for the current behavior, that empty strings are retained in an encoded StatName, which that in-progress PR changes.

So I'll hold off on this PR and wait for that one to settle out.

**Pull request closed without merging**